profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,153 @@
+##############################################################################
+# Input Variables
+##############################################################################
+
+variable "resource_group_id" {
+  description = "ID of resource group to use when creating the mongodb database"
+  type        = string
+}
+
+variable "mongodb_version" {
+  description = "The version of mongodb. If null, the current default ICD mongodb version is used."
+  type        = string
+  default     = null
+}
+
+variable "plan" {
+  type        = string
+  description = "The name of the service plan that you choose for your mongodb instance"
+  default     = "standard"
+}
+
+variable "region" {
+  description = "The region mongodb is to be created on. The region must support BYOK ( us-south, us-east, and eu-de)"
+  type        = string
+  default     = "us-south"
+  validation {
+    condition = anytrue([
+      var.region == "us-south",
+      var.region == "us-east",
```

Comment:
hmm only keys in the `us-south` and `eu-de` are durable to region failures, so should we allow us-east for the fscloud profile?

Reply:
I think yes, as these regions support BYOK. I don't think that we should have a check only for the regions which are durable during failures as this variable defines which location MongoDB should be created.

Reply:
But the mongodb location has to match to the location of the key

Reply:
For FSCloud we need KYOK not BYOK so that is Hyper Protect. Not sure how this affects the regions....